src/main/c/javacan_bcmmessage.c

```diff
@@ -0,0 +1,71 @@
+/**
```

Comment:
I'm not stoked either. I thought i can create the whole message from pure Java. But on the first test on an ARM device i had to learn about padding in C structures.
So this current solution appeared to me to be the most simple and reliable way. Without adding much C code, but still being able to create a valid struct from Java on all platforms.
We have an in-house lib for CAN that creates the struct from the Java object inside the C code. This is nasty code!

In case of the CanFrame you are a bit lucky, as all the struct members are fixed size datatypes (`__u32`, `__8`) and the struct itself already contains the padding. So you are on the safe side.

Reply:
Agreed.